### ISSUE TITLE
cannot run inside flask

### ISSUE THREAD

**Issue by lequangdzung** [invisibleroads/socketIO-client#115]
I tried to run `socketIO-client` inside my flask app, but it stop flask forever.

Here is my code:

```
import sys
from app import create_app
from socketIO_client import SocketIO, BaseNamespace

import logging

class TestNamespace(BaseNamespace):
    def on_aaa_response(self, *args):
        print('on_aaa_response', args)

    def on_test(self, data):
        print "on test"

    def on_connect(self):
        print 'on connect'

if __name__ == '__main__':
    logging.getLogger('requests').setLevel(logging.WARNING)
    logging.basicConfig(level=logging.DEBUG)
    env_name = sys.argv[1] if len(sys.argv) > 1 else 'dev'

    app = create_app(env_name=env_name)
    app.config['DEBUG'] = True
    app.config['SERVER_NAME'] = "localhost:8080"

    socketio = SocketIO('localhost', 5000)
    namespace = socketio.define(TestNamespace, '/test')
    app.run('0.0.0.0', 8080, debug=True)
```

But i got result like this:

```
DEBUG:root:localhost:5000/socket.io [transport selected] websocket
DEBUG:root:localhost:5000/socket.io [heartbeat reset]
DEBUG:root:localhost:5000/socket.io [socket.io packet sent] 0/test,
DEBUG:root:localhost:5000/socket.io [socket.io packet received] 0/test
on connect
```

Then i cannot access my site from `http://localhost:8080`

**Comment by ChaoticMind** [comment]
This is most likely #117

**Comment by invisibleroads** [comment]
1. Your code might be hanging when you are defining the namespace because the namespace is not defined on the server.
2. You need to run SocketIO().wait() in order to process events. Since flask is in the main thread, you will want to put the socketIO-client in a separate thread or separate process. 
3. If you only need to emit events, then you can use SocketIO().wait(seconds=1)

Duplicate of https://github.com/invisibleroads/socketIO-client/issues/84

Fixed in 2abcc91fb7f6c5a5d1da1f5edf2fda1d85890015

Please re-open this issue if it is still a problem.
